### PR TITLE
🐛 Fix self-closing tags

### PR DESCRIPTION
**Pull request by alanorozco**
Some rendering scenarios cause the elements to be nested as opposed to siblings:

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
Agree with @jridgewell on linting for this. 

The void element list is fairly small:
whatwg: https://html.spec.whatwg.org/multipage/syntax.html#void-elements
handy package: https://github.com/wooorm/html-void-elements

**Comment by jridgewell** [comment]
Already have a lint PR coming.

**Comment by jridgewell** [comment]
On second though, I just pushed my commit to this branch (and this PR).

**Comment by timhaines** [comment]
@alanorozco @rsimha I just took a look at the travis build log, and saw the `[15:21:48] Starting 'visual-diff'...` and then Travis kill the build due to lack of output after 10 minutes.  

I'm 99.9% certain that the delay is not due to Percy's processing time.  There's not a lot to go on in the logs that may indicate what the issue is though, so triaging would require some research.

Are those console errors that are reported expected?

@rsimha this is a problem we'd be happy to help with after we get the support contract in place.

File: build-system/eslint-rules/html-template.js [hunk 1]
```diff
@@ -65,6 +65,38 @@ module.exports = function(context) {
...
+          message: `Invalid void tag "${tag}"`,
+        });
+      }
+    }
+  }
+
+  function invalidVoidTag(string) {
+    // Void tags are defined at
+    // https://html.spec.whatwg.org/multipage/syntax.html#void-elements
+    const invalid = /<(?!area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)([a-zA-Z-]+) [^>]+\/>/g;
```
Comment (cvializ) [inline build-system/eslint-rules/html-template.js]: @jridgewell This regex doesn't match all invalid void tags. For example, it doesn't match `<div />` or `<div/>`
https://regexr.com/3ojnm

Comment (jridgewell) [reply build-system/eslint-rules/html-template.js]: I have a fix we can do after this is merged. There are no linting errors even with it.

**Comment by timhaines** [comment]
@rsimha Yes, definitely happy to revisit.

The one bit of extra context around why I don't think Percy's "server processing and rendering" was responsible for the 28 minutes is that I suspect something was holding up the client-run from calling "finalize" in our API after the snapshots have been taken.  Percy won't complete the build until that occurs.  

If you wanted to dig into this a little, you could add some logging to determine when finalize is being called, and I expect you'll see that's taking ~28 minutes from the time the build is started, for some reason.  I may be wrong, but this is my suspicion of what's going on..  The first place I'd look for why that might be occurring (if it is) is why Selenium may be taking a very long time to capture your DOM snapshots.  This is all (experienced) speculation at this point though.

**Comment by rsimha** [comment]
@timhaines Thanks for that info. Here's what I found:

The Travis [logs](https://travis-ci.org/ampproject/amphtml/jobs/371350454#L648-L656) show that the initial `gulp visual-diff` command (that generates and uploads the snapshots) completed successfylly in 1 minute. I checked to make sure that we do [finalize](https://github.com/ampproject/amphtml/blob/master/build-system/tasks/visual-diff.rb#L301) the build towards the end of that task. The Travis build timed out during the subsequent call to `gulp visual-diff --verify`. That step waits until Percy processes the build and reports whether there were diffs or not.

Anyhow, our infrastructure is under a large amount of stress these days, and I'd imagine things will get better once we add capacity throughout the stack.

Thanks, as always, for your prompt and kind help!

File: build-system/eslint-rules/html-template.js (hunk 1 above)
Comment (jridgewell) [reply build-system/eslint-rules/html-template.js]: Build failed again, so I pushed the fix here.

**Pull request merged**